core-test/src/test/scala/org/bitcoins/core/protocol/dlc/ContractDescriptorTest.scala

```diff
@@ -0,0 +1,89 @@
+package org.bitcoins.core.protocol.dlc
+
+import org.bitcoins.core.currency._
+import org.bitcoins.core.protocol.dlc.models._
+import org.bitcoins.core.protocol.tlv.ContractDescriptorV1TLV
+import org.bitcoins.testkitcore.util.BitcoinSUnitTest
+
+class ContractDescriptorTest extends BitcoinSUnitTest {
+
+  behavior of "ContractDescriptor"
+
+  it should "fail to create an empty EnumContractDescriptor" in {
+    assertThrows[IllegalArgumentException](EnumContractDescriptor(Vector.empty))
+  }
+
+  it should "fail for not starting with a endpoint" in {
+    val func = DLCPayoutCurve(
+      Vector(
+        OutcomePayoutPoint(0, Satoshis(0), isEndpoint = false),
+        OutcomePayoutPoint(3, Satoshis(100), isEndpoint = true)
+      ))
+    assertThrows[IllegalArgumentException](
+      NumericContractDescriptor(func, 2, RoundingIntervals.noRounding))
+  }
+
+  it should "fail for not ending with a endpoint" in {
+    val func = DLCPayoutCurve(
+      Vector(
+        OutcomePayoutPoint(0, Satoshis(0), isEndpoint = true),
+        OutcomePayoutPoint(3, Satoshis(100), isEndpoint = false)
+      ))
+    assertThrows[IllegalArgumentException](
+      NumericContractDescriptor(func, 2, RoundingIntervals.noRounding))
+  }
+
+  it should "fail for starting below the minimum" in {
+    val func = DLCPayoutCurve(
+      Vector(
+        OutcomePayoutPoint(-1, Satoshis(0), isEndpoint = true),
+        OutcomePayoutPoint(3, Satoshis(100), isEndpoint = true)
+      ))
+    assertThrows[IllegalArgumentException](
+      NumericContractDescriptor(func, 2, RoundingIntervals.noRounding))
+  }
+
+  it should "fail for starting above the minimum" in {
+    val func = DLCPayoutCurve(
+      Vector(
+        OutcomePayoutPoint(1, Satoshis(0), isEndpoint = true),
+        OutcomePayoutPoint(3, Satoshis(100), isEndpoint = true)
+      ))
+    assertThrows[IllegalArgumentException](
+      NumericContractDescriptor(func, 2, RoundingIntervals.noRounding))
+  }
+
+  it should "fail for starting below the maximum" in {
+    val func = DLCPayoutCurve(
+      Vector(
+        OutcomePayoutPoint(0, Satoshis(0), isEndpoint = true),
+        OutcomePayoutPoint(2, Satoshis(100), isEndpoint = true)
+      ))
+    assertThrows[IllegalArgumentException](
+      NumericContractDescriptor(func, 2, RoundingIntervals.noRounding))
+  }
+
+  it should "fail for starting above the maximum" in {
+    val func = DLCPayoutCurve(
+      Vector(
+        OutcomePayoutPoint(0, Satoshis(0), isEndpoint = true),
+        OutcomePayoutPoint(4, Satoshis(100), isEndpoint = true)
```

Comment:
Functionally the same as the last case?

Reply:
Yes, just making sure we over everything :)

Reply:
Shouldn't it be 104 to cover above the maximum?

Reply:
No, the second param here is the `payout` which can be anything. The min/max refer to the oracle outcome which is the first param here. Since we are using a `numDigits` of `2` here our max value is `(2^2) - 1 = 3`

Reply:
Thank you for explaining